Refactor to svelte

martinezpl opened the pull request:
closes #11
closes #12

a96lex committed:
add boilerplate for svelte+vite+ts

a96lex committed:
update ff manifest

a96lex committed:
migration: define chat structure & copy styles

a96lex committed:
remove unneeded code

a96lex committed:
port more of the functionality to svelte

a96lex committed:
a bit of style

a96lex committed:
update interfaces

a96lex committed:
no need to export these

a96lex committed:
moved img

a96lex committed:
add toggle state store

a96lex committed:
remove placeholder

a96lex committed:
throw error

a96lex committed:
add owly state + images

a96lex committed:
add owly state changes

a96lex committed:
update askQuestion func so we can clear the input immediately

a96lex committed:
add helper to update video time

a96lex committed:
export `TranscriptNode` type

a96lex committed:
update console debug

a96lex committed:
add transcript component

a96lex committed:
nicer font

a96lex committed:
update style a bit

a96lex committed:
removed vanilla code

we can check later in history what may be missing

a96lex committed:
rename file chat -> chatTypes and update imports

a96lex committed:
fix bug in which opening chat scrolls it smoothly

a96lex committed:
get font via svelte

a96lex committed:
let me use icons pls

a96lex committed:
add replace plugin

a96lex committed:
add replace plugin

a96lex committed:
add hack to access `chrome.runtime.getURL(`

a96lex commented:
Changing to WIP. PR needs a couple more things before it's ready

- [x] update/refactor build script
- [x] update server response types
- [x] update readme
  - [x] build instructions
  - [x] development instructions -> TODO: fix dev
- [x] update chrome manifest
- [ ] add tests for server calls -> TODO

A caveat of the migration is this hack: e59777f5fd2dab3b971a4b4ab172a84f53bb33a6.

I asked [a question](https://stackoverflow.com/questions/75258357/how-to-access-chrome-runtime-from-svelte-project) in stackoverflow to see if we can find a better approach.

Commit:
Icon revamp

Commit:
Revert font change

Commit:
Update build script

Commit:
Adjust css

Commit:
Hide toggle on fullscreen

Commit:
Fix ready icon

Commit:
Have ready icon same as steady

Commit:
Fix backend communication

Commit:
Switch message styling (Owlie is on the right side ;p)

Commit:
.

a96lex committed:
remove replace when dev mode

a96lex committed:
use css for global styles

https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/faq.md#where-should-i-put-my-global-styles

a96lex committed:
updated readme

Commit:
Integrate new API, render errors red

Commit:
Add user code handling

Commit:
Use webextension-polyfill for multiplatform handling

Commit:
Include background script in the build

Commit:
Generate owlieId on install, store it and pass it into the request

a96lex committed:
ran prettier

a96lex committed:
update lock

a96lex committed:
remove replacing plugin

Commit:
New beautiful icons

Commit:
Merge branch 'refactor/svelte' of github.com:martinezpl/owlie-yt-addon into refactor/svelte

a96lex committed:
rename app

a96lex committed:
move app and main

a96lex committed:
move some files

a96lex committed:
upadte manifest

note: i  get an error when loading v3 on chrome

a96lex committed:
add popup app

a96lex committed:
add shared dir

localstorage stuff will be here

a96lex committed:
add remaining html

a96lex committed:
add build scripts

a96lex committed:
allow building multiple apps

on successive build commands

a96lex committed:
rename content_scripts in manifest v2

a96lex committed:
no need for switch.

just split at `:` if possible else default to `app`

a96lex committed:
no need to infer here

a96lex committed:
js -> ts

this fixes warnings in tsconfig.json

a96lex committed:
ran prettier, add any type

a96lex committed:
update readme

a96lex reviewed (approved):
PR is too big, time to merge!
